packages/s3-credentials-endpoint/index.js

```diff
@@ -47,183 +36,6 @@ const buildEarthdataLoginClient = () =>
- *  }
- *
- * @param {Array<string>} cmrAllowedBucketKeyList - earthdata login user name
- * @returns {string} - stringified payload for policy helper function.
- */
-function formatAllowedBucketKeys(cmrAllowedBucketKeyList) {
-  const bucketKeyPairList = cmrAllowedBucketKeyList.map(parseBucketKey);
-  const bucketlist = [];
-  const pathlist = [];
-
-  bucketKeyPairList.forEach((bucketKeyPair) => {
-    bucketlist.push(bucketKeyPair.bucket);
-    pathlist.push(bucketKeyPair.keypath);
-  });
-
-  return JSON.stringify({
-    accessmode: 'Allow',
-    bucketlist,
-    pathlist,
-  });
-}
-
-/**
- *  Retrieve the sts session policy for a user when s3 credentials endpoint is
- *  configured to use CMR ACLs. If the endpoint is not configured for CMR ACLs,
- *  return undefined.
- *
- * @param {string} edlUser - earthdatalogin username
- * @param {string} cmrProvider - Cumulus' CMR provider.
- * @param {Object} lambda - aws lambda service.
- * @returns {Object} session policy generated from user's CMR ACLs or undefined.
- */
-async function fetchPolicyForUser(edlUser, cmrProvider, lambda) {
-  if (!configuredForACLCredentials()) return undefined;
-
-  // fetch allowed bucket keys from CMR
-  const cmrAllowedBucketKeyList = await getUserAccessibleBuckets(edlUser, cmrProvider);
-  const Payload = formatAllowedBucketKeys(cmrAllowedBucketKeyList);
-
-  return lambda.invoke({
-    FunctionName: process.env.STS_POLICY_HELPER_LAMBDA,
-    Payload,
-  }).promise()
-    .then((lambdaReturn) => JSON.parse(lambdaReturn.Payload));
-}
-
-/**
- * Dispenses time-based temporary credentials for same-region direct s3 access.
- *
- * @param {Object} req - express request object
- * @param {Object} res - express response object
- * @returns {Object} the express response object with object containing
- *                   tempoary s3 credentials for direct same-region s3 access.
- */
-async function s3credentials(req, res) {
-  ensureEndpointEnabled(res);
-
-  const roleSessionName = buildRoleSessionName(
-    req.authorizedMetadata.userName,
-    req.authorizedMetadata.clientName
-  );
-
-  const policy = await fetchPolicyForUser(
-    req.authorizedMetadata.userName,
-    process.env.cmr_provider,
-    req.lambda
-  );
-
-  const credentials = await requestTemporaryCredentialsFromNgap({
-    lambda: req.lambda,
-    lambdaFunctionName: process.env.STS_CREDENTIALS_LAMBDA,
-    userId: req.authorizedMetadata.userName,
-    roleSessionName,
-    policy,
-  });
-
-  const creds = JSON.parse(credentials.Payload);
-  if (Object.keys(creds).some((key) => ['errorMessage', 'errorType', 'stackTrace'].includes(key))) {
-    log.error(credentials.Payload);
-    return res.boom.failedDependency(
-      `Unable to retrieve credentials from Server: ${credentials.Payload}`
-    );
-  }
-  return res.send(creds);
-}
-
 // Running API locally will be on http, not https, so cookies
 // should not be set to secure for local runs of the API.
 const useSecureCookies = () => {
```

Comment:
You can use the function from https://github.com/nasa/cumulus/blob/master/packages/api/lib/distribution.js#L11

Reply:
Also `bucketNameFromPath` and `isPublicRequest` are defined in https://github.com/nasa/cumulus/blob/master/packages/api/app/distribution-routes.js
You can move them to
https://github.com/nasa/cumulus/blob/master/packages/api/lib/distribution.js

I feel we should remove the duplicate code if the function is already defined in api package

Reply:
Moved 👍